akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -719,9 +724,11 @@ abstract class ShardCoordinator(
           aliveRegions += region
           if (state.regions.contains(region)) {
             region ! RegisterAck(self)
+            informAboutCurrentShards(region)
             allocateShardHomesForRememberEntities()
           } else {
             gracefulShutdownInProgress -= region
+            informAboutCurrentShards(region) // we don't have to wait for updated state for this
```

Comment:
I thought this was a good idea because with this a health check could wait for the region to have gotten the register ack and be up to speed when it starts accepting requests, but on the other hand, if there is a huge number of shards, then maybe delaying RegisterAck to after all those shards has been sent/handled is a bad idea?

Reply:
@patriknw wdyt abut this, should we add an upper cap to how many shards we push before register Ack and push the rest afterwards?

Reply:
I'm not sure I understand the problem? Sending the ShardHomes immediately sounds good to me.

By default we have `number-of-shards = 1000` so that would be 2 messages. Are you concerned that someone has configured like 100k shards? If that is a concern we should have a cap in informAboutCurrentShards and not send them at all if too many.

Reply:
Yeah, something like that was my thinking, perhaps I'm over-concerned, say something more normal like 10 000 shards with 10 bytes shard id per shard, with the region as well and overhead that'd be just above ~100k, split up over 20 messages. Maybe not a huge problem.

Reply:
The batch size is 500 so 100,000 shards would be split into 200 ShardHomes messages, or did I misunderstand something?

I think it would be good to add an upper bound of 10 messages in informAboutCurrentShards

Reply:
Alright, I'll add.